docs_espressif/en/additionalfeatures/coverage.rst

```diff
@@ -0,0 +1,70 @@
+
+.. image:: ../../../media/tutorials/basic_use/gui_menuconfig.png
+
+The example will enable the following options by default:
+
+- Enable the Application Tracing Module under ``Component Config`` -> ``Application Level Tracing`` -> ``Data Destination`` by choosing `Trace Memory`.
+- Enable GCOV to host interface under ``Component Config`` ->`` Application Level Tracing`` -> ``GCOV to Host Enable``.
+- Enable OpenOCD Debug Stubs under ``Component Config`` -> ``ESP32-specific`` -> ``OpenOCD Debug Stubs``
+
+Now to build the project, flash your device and start the ESP-IDF Monitor you can use the **ESP-IDF: Build your Project**, **ESP-IDF: Flash your Project** and **ESP-IDF: Monitor Device** commands.
+
+.. note::
+  There is also a **ESP-IDF: Build, Flash and Start a Monitor on your Device** command which does all three former commands in one.
+
+Next step is to launch OpenOCD and send some commands. To start openOCD from the extension, execute the **ESP-IDF: OpenOCD Manager** command or from the ``OpenOCD Server (Running | Stopped)`` button in the Visual Studio Code status bar. OpenOCD server output is shown in menu **View** -> **Output** -> **ESP-IDF**.
+
+Launch a new terminal with menu Terminal -> New Terminal and execute ``telnet <oocd_host> <oocd_port>`` which by default is ``telnet localhost 4444``. Latest macOS users can use ``nc <oocd_host> <oocd_port>`` if ``telnet`` is not in the system.
+
+.. note::
+  The user can modify **openocd.tcl.host** and **openocd.tcl.port** configuration settings to modify these values.
+
+Send the openOCD command ``esp gcov dump`` for hard-coded dump which will dump two hard-coded dumps based on this example. After that send the ``esp gcov`` command for instant run-time dump.
+
+.. image:: ../../../media/tutorials/coverage/oocd_cmds.png
+
+After dumping data one or more times, open the desired file in your editor and execute the **ESP-IDF: Add Editor Coverage** command to highlight the editor with code coverage.
+
+You can customize highlight color using these extension settings.json configuration settings:
+
+- Covered lines use **idf.coveredLightTheme** for light themes and **idf.coveredDarkTheme** for dark themes.
+- Partially covered lines use **idf.partialLightTheme** for light themes and **idf.partialDarkTheme** for dark themes.
+- Non-covered lines use **idf.uncoveredLightTheme** for light themes and **idf.uncoveredDarkTheme** for dark themes.
+
+Visual Studio code support ``"red"``, ``rgb(255,0,120)`` or ``rgba(120,0,0,0.1)``.
+
+.. image:: ../../../media/tutorials/coverage/editor_coverage.png
+
+When finished, use the **ESP-IDF: Remove Editor Coverage** command to remove the code coverage.
+
+The user can generate a html report using the **ESP-IDF: Get HTML Coverage Report for Project** command.
+
+.. image:: ../../../media/tutorials/coverage/html_report.png
+
+.. warning::
```

Comment:
I feel like this should note be a warning, rather a Note or Important tag should be enough